src/acceptance/assets/app/go_app/internal/app/custom_metrics.go

```diff
@@ -94,7 +94,7 @@ func (_ *CustomMetricAPIClient) PostCustomMetric(ctx context.Context, appConfig
 		return err
 	}
 
-	apiClient, err := api.NewClient(autoscalerApiServerURL, api.WithClient(client))
+	apiClient, err := api.NewClient(autoscalerApiServerURL, autoscalerCredentials, api.WithClient(client))
```

Comment:
Is this necessary? Does it mean that even in the mTLS case we now supply basic auth credentials?
For code clarity sake, I would then rather prefer to put the `api.NewClient` calls in the `if useMtls` branches above.

Reply:
The generated code now always includes a field in the client-struct for basic authentication. Without modifying the generated code, we really need to use it here. Perhaps when we drop basic authentication, it will go.